Blocks: Update keywords for better search results

obenland opened the pull request:
Changes and adds keywords to a few blocks that will hopefully be easier to find now.

Fixes #14628.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Updates `map` keyword to `maps` in Map block. This will allow for both strings to return a result in the block selector.
* Adds `social` keyword to Pinterest block.
* Adds a variety of new keywords to some core blocks, hoping for better findability.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
See Quick Wins in paYJgx-qn-p2.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Run `yarn build-extensions` to compile these changes.
* Open the Editor, then open the block picker and search for "social". Facebook, Twitter, Tumblr, Pinterest, and Instagram blocks should be part of the search results.
* In your browser console, run `wp.blocks.getBlockType('core-embed/crowdsignal')` and check to see that the "survey" keyword was added.

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* Enhanced block search results through additional keywords.

obenland committed:
Blocks: Update keywords for better search results

Changes and adds keywords to a few blocks that will hopefully be easier to find now.

marekhrabe commented on extensions/shared/block-keywords.js:
```diff
@@ -0,0 +1,46 @@
+			if ( ! includes( settings.keywords, video ) ) {
+				settings.keywords.push( video );
+			}
+			break;
```
Should we open a core editor issue to add these keywords? They make more sense to me there but it's nice we can test them sooner with this 👍

jeherve replied:
This definitely feels like a better fit for Core overall. We can add this in Jetpack for now, but it would be nice if we had a clear path to deprecating this (e.g. a matching Gutenberg PR and a comment here inviting the removal of the filter once we only support WP 5.4+, for example).

obenland replied:
I added a file comment with a reference

marekhrabe commented on extensions/blocks/map/settings.js:
```diff
@@ -26,7 +26,7 @@ export const settings = {
 	category: 'jetpack',
 	keywords: [
-		_x( 'map', 'block search term', 'jetpack' ),
+		_x( 'maps', 'block search term', 'jetpack' ),
```
This one is interesting, as the trick with "map" and "maps" matching because one word fully contains the other only works in the English (possibly in some other languages too but it's not an universal rule). 

Should we just list both variants as there is no longer a keyword limit?

jeherve replied:
> there is no longer a keyword limit

Is that a new change in Core? We may need to keep only 3 for now since we support WP 5.2 for a bit longer in Jetpack.

marekhrabe replied:
the limit was removed in February 2019 - https://github.com/WordPress/gutenberg/pull/13848 (follow the discussion as it brings more context than just the description)

jeherve replied:
That's good, Gutenberg 5.1 was released a while ago now so that's definitely something we can use in Jetpack today.

marekhrabe reviewed (dismissed):
I have left two comments but I wouldn't mind if this PR got in as is. Already a big improvement and we can (and will) continue making changes in this area further. I've also tested the wpcom diff and it matches described changes

obenland committed:
Make `maps` a separate keyword

Just because `map` is part of `maps` in English, doesn't mean that that'll be the case in all languages.

Props @marekhrabe

obenland committed:
Add core PR reference.

Props @jeherve.

obenland commented:
@jeherve I'm not sure who's in charge of `github-phab-bot` but (1) that automation is pretty cool, unfortunately (2) it doesn't seem to sync over added files 😬

Thanks for your review, I think this is ready for another round.

obenland committed:
Remove block keywords filter

We'll use core's update

jeherve commented:
> 1) that automation is pretty cool

It is :) The tool is part of Fusion, which helps us keep Jetpack and wpcom in sync.

> unfortunately (2) it doesn't seem to sync over added files

Correct. That's something we have yet to implement. For now, that means that once this PR is merged here in Jetpack, you'll need to copy the file over to your sandbox manually, `svn add` it, and update the Fusion-generated diff so it includes that file (in practice that will mean something like:

```
arc diff --update D38792 xx \
xx \
xx
```

xx being all the unbuilt files you've modified / added. `github-phab-bot` will then come around and update your diff to add all built files to the diff for you.
Let me know if you need help with any of this!
